### PR TITLE
Added is_valid_block() and associated test

### PR DESCRIPTION
**Pull request by jldistefano**

### CODE REVIEW

**Comment by kstola2** [comment]
There's a lot of guidance I tried to put in the issue description, namely in the Testing portion.

**Comment by jldistefano** [comment]
Requested changes have been made and my branch has been updated.

File: data_structures/block.py [hunk 1]
```diff
@@ -389,3 +389,23 @@ def parse_block(block_header):
...
+    :param block: 74 byte string representing the previous block in the blockchain. output of mine()
+    :returns: boolean True if all the above conditions are met, False otherwise
+    """
+    block_info = parse_block(block)
+    prev_block_info = parse_block(prev_block)
+    if (block_info["timestamp"] <= prev_block_info["timestamp"]):
+        return False
+    if (block_info["prev_hash"] != hash_SHA(prev_block)):
+        return False
+    if (block_info["target"] <= hash_SHA(block)):
```
Comment (kstola2) [inline data_structures/block.py]: Can you switch this to be hash_SHA(block) > block_info['target'] I think it's more explicit that way.

File: data_structures/block.py [hunk 2]
```diff
@@ -389,3 +389,23 @@ def parse_block(block_header):
     parsed_block["target"] = slice_target(block_header)
     parsed_block["nonce"] = slice_nonce(block_header)
     return parsed_block
+
+def is_valid_block(block, prev_block):
+    """
+    Compares a block and its previous block to determine if block is allowed to be added to blockchain
+    Confirms that the timestamp of block is larger than that of prev_block
+    Confirms that prevHash member of block is equal to hash of prev_block
+    Confirms that target of block is greater than target of prev_block
```
Comment (kstola2) [inline data_structures/block.py]: This line needs to be done away with because it's not what we're testing.

File: data_structures/block.test.py [hunk 3]
```diff
@@ -266,7 +266,47 @@ def test_parse_block(self):
...
+        #Creates a valid previous block
+        prev_block = mine(hash_SHA("0".encode()), data, target)
+        prev_hash = hash_SHA(prev_block)
+        #Creates a valid block
+        block = mine(prev_hash, hash_SHA("2".encode()), target)
+
+        #Creates an invalid block with invalid prev_hash
+        inv_block_2 = mine(inv_prev_hash, hash_SHA("INVALID".encode()), target)
+        
+        #Creates an invalid block with block hash less than its target
```
Comment (kstola2) [inline data_structures/block.test.py]: Blocks with hashes less than it's target constitute a valid block.

**Pull request closed without merging**